.github/workflows/soundness.yml

```diff
@@ -177,3 +185,21 @@ jobs:
         run: |
           apt-get -qq update && apt-get -qq -y install shellcheck
           git ls-files -z '*.sh' | xargs -0 shellcheck
+
+  yaml-lint:
+    name: Lint YAML
+    if: ${{ inputs.yamllint_enabled }}
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ inputs.yamllint_container_image }}
+    timeout-minutes: 5
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Run yamllint
+        run: |
+          apt-get -qq update && apt-get -qq -y install yamllint
```

Comment:
The runner itself comes with yamlint preinstalled so we don't have to install it here